### ISSUE TITLE
wtf is that

### ISSUE THREAD

**Issue by ap0sentada** [Marak/colors.js#290]
when i run code with `const color = require("colors")` i recive this log
![image](https://user-images.githubusercontent.com/70923059/148704826-598185e3-828d-49ec-919d-14e77448bcd1.png)
![image](https://user-images.githubusercontent.com/70923059/148704841-ad086f78-385b-41e7-9e52-8810fc456d1b.png)

**Comment by dustinlw1987** [comment]
Marak is a greedy terrible person that decided to severely screw over his users by introducing malicious code that intentionally breaks colors.js.
https://www.bleepingcomputer.com/news/security/dev-corrupts-npm-libs-colors-and-faker-breaking-thousands-of-apps/

He also blew up his apartment and apparently beat up his girlfriend.
https://abc7ny.com/suspicious-package-queens-astoria-fire/6425363/

In other words, do not trust anything from this person.

**Comment by DABH** [comment]
This is a duplicate of #285 .  For sake of making solutions easy to find, please consider closing this issue so folks go to #285 instead.  Thanks.

**Comment by dustinlw1987** [comment]
Marak should NOT be trusted as a developer! Especially after doing this unilaterally without notification.

**Comment by roberrrt-s** [comment]
> Marak is a greedy fuck that decided to fuck over his users by introducing malicious code that intentionally breaks colors.js.

You're using someone else's free work and call them greedy fucks?

**Comment by roberrrt-s** [comment]
@dustinlw1987 You're literally calling someone that helped millions of developers a greedy fuck what is wrong with you lol.

**Comment by dustinlw1987** [comment]
> @dustinlw1987 You're literally calling someone that helped millions of developers a greedy fuck what is wrong with you lol.

I'm going to call him out for his own actions. He burnt bridges and destroyed any possible trust anybody could have for him.

Fucking not sorry.

**Comment by will-holley** [comment]
If you're using yarn, you can resolve this issue by adding the following to your `package.json`:

```
"resolutions": {
  "colors": "1.4.0"
}
```

**Comment by roberrrt-s** [comment]
> > @dustinlw1987 You're literally calling someone that helped millions of developers a greedy fuck what is wrong with you lol.
> 
> I'm going to call him out for his own actions. He burnt bridges and destroyed any possible trust anybody could have for him.
> 
> Fucking not sorry.

It's always our responsibility to make sure we're not using malicious code. Always.

**Comment by dustinlw1987** [comment]
> 

Precisely! It is also our responsibility to call out malicious actions.

**Comment by shayneoneill** [comment]
Goddamn. I took down our whole infrastructure at work thinking we had been hacked.

Did he think for a second that this was hitting back at the big guys? It wasn't . It just fucked over a lot of us 
guys who are regular working slobs who now have to explain to the boss why our deployments imploded.

**Comment by seho-dev** [comment]
Fortunately I use chalk.js

**Comment by ChristopherTrimboli** [comment]
Just a reminder that if your project broke because of this, you are using deps unsafely and most likely using a `^` in your package.json versions. In the world of FOSS, this maintainer is free to publish any version they want. Since it is their repository. Beyond the politics or the protest or the broken builds... there exists engineering solutions for this to never effect you and I don't think the solution is to censor or cancel @Marak. 

My shit broke too but I also respect that he is free to do this if he wishes. Don't complain that this is broken when you could write your own color.js and haven't paid to maintain Marek's. We are owed nothing and anything we are given in the FOSS space is given on an honor system at best.

**Comment by kkm** [comment]
can someone fork?

**Comment by hakube** [comment]
> > @dustinlw1987 You're literally calling someone that helped millions of developers a greedy fuck what is wrong with you lol.
> 
> I'm going to call him out for his own actions. He burnt bridges and destroyed any possible trust anybody could have for him.
> 
> Fucking not sorry.

Like he cares. I get his point. Big corporations are leeching off opensource and not give back. I'm not so sure why you called him greedy. There are tons of corporations that uses this library and didn't give back a single penny to support the developer.

People like you and corporations only cares about the product and gives 0 fucks about the developer. In the end, it bit you in the ass

**Comment by jshor** [comment]
~~I've forked this and will maintain `colors` from now on. See #292~~

Edit: @DABH will maintain a fork for this.

**Comment by DABH** [comment]
If you read #285 you’ll see I’ve already forked and am working to resolve the incident with the relevant parties.

**Comment by VisZhangrong** [comment]
### shit

**Comment by sanishchirayath1** [comment]
I am not here to justify what he did... But, the guy is going through a tough period,, His house burned down...He is literally homeless...Big companies are not contributing to the work open source do... It should be a wake-up call to all companies who are using open source code to generate revenues ... consider start contribute to open-source to keep them open-source innovating...

**Comment by ShaofeiZi** [comment]
what？

**Comment by jcschmidig** [comment]
Marak crossed a line here. This seriously damages the whole community. If he wants to be paid directly there are sure other possibilities. This should not happen to anyone.

**Comment by tswordyao** [comment]
> I am not here to justify what he did... But, the guy is going through a tough period,, His house burned down...He is literally homeless...Big companies are not contributing to the work open source do... It should be a wake-up call to all companies who are using open source code to generate revenues ... consider start contribute to open-source to keep them open-source innovating...

it is a terrible reason

**Comment by harish2704** [comment]
@sanishchirayath1 :  FYI, https://abc7ny.com/suspicious-package-queens-astoria-fire/6425363/

**Comment by sanishchirayath1** [comment]
> > I am not here to justify what he did... But, the guy is going through a tough period,, His house burned down...He is literally homeless...Big companies are not contributing to the work open source do... It should be a wake-up call to all companies who are using open source code to generate revenues ... consider start contribute to open-source to keep them open-source innovating...
> 
> it is a terrible reason

Yes, It is...Maybe he is going through a mental breakdown. What he did is wrong no doubt about it.

**Comment by ghost** [comment]
Ahh the cope in the comments is beautiful. So much third party dependency whining and little action taken, so much to be expected from people who only know how to write `npm install` and copy documentation snippets.

If you don't like it, make your own faker.js

If you cannot...then why are you a programmer? Leave and make something else, you **clearly** aren't suited for technology, consider dumpster diving tho, it might be fun :D

**Comment by jcschmidig** [comment]
@VentGrey you are missing clearly the point. It's about trust in the community.

**Comment by cronlabspl** [comment]
> can someone fork?

Sure. Ive been thinking of forking for 2 last days. :) Will be happy to become new maintainer.

**Comment by kimshinoh** [comment]
NISU! waste my time

**Comment by cronlabspl** [comment]
> NISU!

What?

**Comment by cronlabspl** [comment]
> People are still using Marak's projects? Damn.

Ive forked it, detached it from Marak's original work, and am now working on my own version. If you feel need to cooperate, than [feel free to do so](https://github.com/labspl/colors.js)

**Comment by karikera** [comment]
Duplicate of #289

**Comment by euberdeveloper** [comment]
Hi, this is a duplicate of my comment in the other issue

Hi, in case the author will not maintain this project or fix this, I made a fork that restores the normal behaviour. I am not sure if I will maintain this package, but just in case the problem will not be solved and you don't want to be scared when running `npm update`...

https://github.com/euberdeveloper/colors.js

In any case I could also suggest passing to the chalk package, which is very good and has a serious and reliable author

**Comment by minhlucvan** [comment]
It's took me days to debug this issue :'(

**Comment by cronlabspl** [comment]
@minhlucvan its nothing complicated really....... https://github.com/labspl/colors.js/commit/17132b83540c9d0399b2c91ebd6630cc5f9fc4f9

**Comment by cronlabspl** [comment]
@minhlucvan its nothing complicated really....... https://github.com/labspl/colors.js/commit/17132b83540c9d0399b2c91ebd6630cc5f9fc4f9

**Comment by ddzy** [comment]
> If you're using yarn, you can resolve this issue by adding the following to your `package.json`:
> 
> ```
> "resolutions": {
>   "colors": "1.4.0"
> }
> ```

Nice work!

**Comment by cronlabspl** [comment]
> > If you're using yarn, you can resolve this issue by adding the following to your `package.json`:
> > ```
> > "resolutions": {
> >   "colors": "1.4.0"
> > }
> > ```
> 
> Nice work!

Thats **obvious part**, rather than just nice 👍

**Comment by Chester97** [comment]
Do we have any fix for npm?

**Comment by jcschmidig** [comment]
> Do we have any fix for npm?

npm already reverted to 1.4.0

**Comment by shayneoneill** [comment]
"My shit broke too but I also respect that he is free to do this if he wishes. Don't complain that this is broken when you could write your own color.js and haven't paid to maintain Marek's. We are owed nothing and anything we are given in the FOSS space is given on an honor system at best."

Why would you respect someone that doesn't respect you? Nobody is asking anyone to maintain it. Like half the crumbling JS infrastructure, shit gets abandoned. But sabotaging tens of thousands of deployments intentionally is a *very* different matter.

Its like the cock smokers who hack peoples sites and then blame the victim because they didnt understand an insanely complicated subsystem, its just making excuses for what is fundamentally a malevolent and destructive action that hurts other people for what?

This isnt free-software ideology, its sociopathy.

The worst part is most people who got this implosion never actually did anything to deserve it, or even put it in their packages file, its just there because some dependency of a dependency insists on it.

**Comment by nameofSEOKWONHONG** [comment]
나는 기사를 읽었고 이 행위에 대하여 이해한다.
자신의 시간을 투자하여 만든 프로젝트를 망치는 자유도 저자에게 있다고 생각한다.
포춘 500대 기업에서 유명 오픈 소스 프로젝트를 지원하지 않는다는 사실에 더 큰 충격이다.
오픈 소스가 돈을 구걸하지 않는다고 하지만, 예술가가 가난에 허덕이면서 오픈 소스에 투자할 이유는 없다고 생각한다.
피해를 입은 기업에는 유감이지만, 이 사건이 사고의 전환이 되기를 바란다.

**Comment by cronlabspl** [comment]
> 나는 기사를 읽었고 이 행위에 대하여 이해한다.
> 자신의 시간을 투자하여 만든 프로젝트를 망치는 자유도 저자에게 있다고 생각한다.
> 포춘 500대 기업에서 유명 오픈 소스 프로젝트를 지원하지 않는다는 사실에 더 큰 충격이다.
> 오픈 소스가 돈을 구걸하지 않는다고 하지만, 예술가가 가난에 허덕이면서 오픈 소스에 투자할 이유는 없다고 생각한다.
> 피해를 입은 기업에는 유감이지만, 이 사건이 사고의 전환이 되기를 바란다.

Used Google Translate to know whats written here, but I must say that - although I agree with the part of author's free will as of project - I strongly disagree that author of project this size is entitled to ruin thousands other projects that has `color.js` as a dependency........

**Comment by nameofSEOKWONHONG** [comment]
> > 나는 기사를 읽었고 이 행위에 대하여 이해한다.
> > 자신의 시간을 투자하여 만든 프로젝트를 망치는 자유도 저자에게 있다고 생각한다.
> > 포춘 500대 기업에서 유명 오픈 소스 프로젝트를 지원하지 않는다는 사실에 더 큰 충격이다.
> > 오픈 소스가 돈을 구걸하지 않는다고 하지만, 예술가가 가난에 허덕이면서 오픈 소스에 투자할 이유는 없다고 생각한다.
> > 피해를 입은 기업에는 유감이지만, 이 사건이 사고의 전환이 되기를 바란다.
> 
> Used Google Translate to know whats written here, but I must say that - although I agree with the part of author's free will as of project - I strongly disagree that author of project this size is entitled to ruin thousands other projects that has `color.js` as a dependency........
> 
> I also agree with @shayneoneill above: what @Marak did is sociapathy; if @Marak wanted no longer to maintain `colors.js` he could have created an issue here and announce his plans this way, rather than making all this fuss.

내 의견은 아래의 뉴스에 기반한다.
https://www.bleepingcomputer.com/news/security/dev-corrupts-npm-libs-colors-and-faker-breaking-thousands-of-apps/
위 행위에 완전히 동의하지는 않는다.
하지만 오픈 소스의 자유의지와 의무가 누구에게 있는가?
사용자? 창작자?
나는 창작자에 동의하고자 한다.

**Comment by cronlabspl** [comment]
@nameofSEOKWONHONG yes, Ive read this article also.

All the duties were at @Marak's discrecy as he is original author / creator of `color.js`

**Comment by jb7959** [comment]
Wow, I think if the company deeply understands OSS’s philosophy, they would contribute and help this project. But, unfortunately, many companies managers don't know well. 
I understand your feeling, But I can not understand what you did. you just make one of the cases for ‘Why we don’t need to support OSS’

Regardless of what you’ve done, I’m appreciating it you did and your contribution to NPM world

**Comment by cronlabspl** [comment]
@jb7959
> I can not understand what you did

What are you referring to?

**Comment by jb7959** [comment]
> @jb7959
> 
> > I can not understand what you did
> 
> What are you referring to?

- https://snyk.io/blog/open-source-npm-packages-colors-faker/?fbclid=IwAR2n2XkRzFR4GJMpBp5LofVmMZuJhna4KUP3zG2nrhRRS8kW9nem_KvdNDc

**Comment by ChristopherTrimboli** [comment]
> "My shit broke too but I also respect that he is free to do this if he wishes. Don't complain that this is broken when you could write your own color.js and haven't paid to maintain Marek's. We are owed nothing and anything we are given in the FOSS space is given on an honor system at best."
> 
> Why would you respect someone that doesn't respect you? Nobody is asking anyone to maintain it. Like half the crumbling JS infrastructure, shit gets abandoned. But sabotaging tens of thousands of deployments intentionally is a _very_ different matter.
> 
> Its like the cock smokers who hack peoples sites and then blame the victim because they didnt understand an insanely complicated subsystem, its just making excuses for what is fundamentally a malevolent and destructive action that hurts other people for what?
> 
> This isnt free-software ideology, its sociopathy.
> 
> The worst part is most people who got this implosion never actually did anything to deserve it, or even put it in their packages file, its just there because some dependency of a dependency insists on it.

Yes I agree it is on the more dark side of chaotic good personality types... I think Marek was using this as protest which is sometimes done like this to disrupt society in a way where people actually have to listen and pay attention because let's be honest now days there is little room for conversation without action. Cancel culture actually created this situation.

**Comment by jcschmidig** [comment]
> Yes I agree it is on the more dark side of chaotic good personality types... I think Marek was using this as protest which is sometimes done like this to disrupt society in a way where people actually have to listen and pay attention because let's be honest now days there is little room for conversation without action. **Cancel culture actually created this situation**.

This is too easy. Behave as a berserk erases any good argument.
